lib/pages/contacts/data/datasource/device_contact_data_source.dart

```diff
@@ -0,0 +1,24 @@
+import 'package:fluffychat/entity/contact/contact.dart';
+import 'package:flutter_contacts/flutter_contacts.dart' as device_contacts;
+import 'package:fluffychat/pages/contacts/data/datasource/contact_data_source.dart';
+
+
+class DeviceContactDataSource implements ContactDataSource {
```

Comment:
you should call the file device_contact_data_source_impl.dart so we can distinguish from astract classes

Reply:
okay. that's correct.